### PR TITLE
Fixes 1580: Add API to list and fetch tasks

### PR DESCRIPTION
**Pull request by dpang314**
## Summary

Adds API to list tasks:

`GET /api/content-sources/v1.0/tasks/`

and to fetch a task:

`GET /api/content-sources/v1.0/tasks/:uuid`

It defines a GORM model that is shared between the queue and DAO. The DAO only uses GORM to read in tasks.

## Testing steps

Start an introspection task

Make a GET request to `/api/content-sources/v1.0/tasks/` to see all tasks

Make a GET request to `GET /api/content-sources/v1.0/tasks/:uuid` with the task UUID to the details of the specific task

### CODE REVIEW

**Comment by swadeley** [comment]
Hi, rebase please.

**Comment by rverdile** [comment]
oktotest

File: pkg/api/task_info.go [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+	EndedAt   string `json:"ended_at"`   // Timestamp task ended running at
+	Error     string `json:"error"`      // Error thrown while running task
+	OrgId     string `json:"org_id"`     // Organization ID of the owner
+}
+
+type TaskInfoCollectionResponse struct {
+	Data  []TaskInfoResponse `json:"data"`  //Requested Data
+	Meta  ResponseMetadata   `json:"meta"`  //Metadata about the request
+	Links Links              `json:"links"` //Links to other pages of results
+}
```
Comment (swadeley) [inline pkg/api/task_info.go:17]: I think comments starting // should have a space after //

https://go.dev/doc/effective_go#commentary

**Comment by swadeley** [comment]
For QE, these will be the command formats for testing:

`app.content_sources.rest_client.tasks_api.list_tasks()`
`app.content_sources.rest_client.tasks_api.get_task('8a625405-<snip>c71e674640')`

**Review by rverdile — COMMENTED** [review]
Looks really good overall! A few small comments about things to rename. 

Still need to test and that may inspire some more feedback, but I don't anticipate any huge changes.

**Review by rverdile — COMMENTED** [review]
Just have to add the endpoints to here and this is good to go!

https://github.com/content-services/content-sources-backend/blob/main/pkg/middleware/rbac_map.go#L22-L36

**Review by rverdile — APPROVED** [review]
nice job!

**Comment by swadeley** [comment]
Hi David, please rebase again and ping me when its ready for testing.

**Comment by swadeley** [comment]
Hi

Looks good:

`$ curl  -u "$BASIC_AUTH" -X GET "https://${URL}/api/content-sources/v1.0/tasks/"
{"data":[{"uuid":"7a3705bd-54d4-497b-a5c9-9c6fa2c8f882","status":"completed","created_at":"2023-06-05T18:19:37Z","ended_at":"2023-06-05T18:19:37Z","error":"","org_id":"3340851"},{"uuid":"dfedeabd-9f26-44fc-b5e2-ce5422f2bf2d","status":"completed","created_at":"2023-06-05T17:59:29Z","ended_at":"2023-06-05T17:59:30Z","error":"","org_id":"3340851"},{"uuid":"35372449-83e8-49c4-abc8-e2854a51ce7f","status":"completed","created_at":"2023-06-05T17:56:56Z","ended_at":"2023-06-05T17:57:02Z","error":"","org_id":"3340851"}],"meta":{"limit":100,"offset":0,"count":3},"links":{"first":"/api/content-sources/v1.0/tasks/?limit=100\u0026offset=0","last":"/api/content-sources/v1.0/tasks/?limit=100\u0026offset=0"}}`


`$ curl  -u "$BASIC_AUTH" -X GET "https://${URL}/api/content-sources/v1.0/tasks/7a3705bd-54d4-497b-a5c9-9c6fa2c8f882"
{"uuid":"7a3705bd-54d4-497b-a5c9-9c6fa2c8f882","status":"completed","created_at":"2023-06-05T18:19:37Z","ended_at":"2023-06-05T18:19:37Z","error":"","org_id":"3340851"}`

**Pull request merged**